python/ray/dag/compiled_dag_node.py

```diff
@@ -843,32 +893,45 @@ def _get_or_compile(
                     #     print(result)
 
                     #     compiled_dag.teardown()
-                    reader_handles = [self._driver_actor]
+                    assert self._actor_handle is not None
+                    reader_to_node.append(
+                        (self._actor_handle, self._get_node_id(self._actor_handle))
+                    )
                 else:
-                    reader_handles = [
-                        reader.dag_node._get_actor_handle() for reader in readers
-                    ]
+                    for reader in readers:
+                        reader_handle = reader.dag_node._get_actor_handle()
+                        reader_to_node.append(
+                            (reader_handle, self._get_node_id(reader_handle))
+                        )
+
                 fn = task.dag_node._get_remote_method("__ray_call__")
                 task.output_channel = ray.get(
                     fn.remote(
                         do_allocate_channel,
-                        reader_handles,
+                        reader_to_node,
                         typ=type_hint,
                     )
                 )
                 actor_handle = task.dag_node._get_actor_handle()
                 self.actor_refs.add(actor_handle)
                 self.actor_to_tasks[actor_handle].append(task)
             elif isinstance(task.dag_node, InputNode):
+                reader_to_node: List[Tuple["ray.actor.ActorHandle", str]] = []
+                # TODO (kevin85421): We need to remove reader_handles_set because
+                # readers can be on the same actor.
```

Comment:
Can you make the TODO clearer? e.g., adding information why this is no longer a constraint

Reply:
Updated https://github.com/ray-project/ray/pull/46700/commits/3a9b6b8a9833a4d46c5dbba8b28567d6197b0955